components/skus/browser/rs/cxx/src/lib.rs

```diff
@@ -125,8 +124,16 @@ mod ffi {
         Error = 4,
     }
 
+    impl UniquePtr<SkusResult> {}
```

Comment:
Is this necessary? Is it sufficient to make sure the dtor runs on the same side of ffi as the ctor?

Reply:
This is the only way that UniquePtr worked. I wasn't able to instantiate a UniquePtr<SkusResult> without this.

Reply:
Right, but that's still talking about rust-side bindings for C++ types. `SkusResult` originates on the Rust side. I find the documentation ambiguous here, in that it says the object wrapped by the UniquePtr is instantiated "on the heap" but it doesn't say *which* heap.

I would be surprised if it did the wrong thing, but wasn't able find a clear example tracing through the code that the correct destructor will be called from the C++ side.

Reply:
Hey, I got rid of this now. We're able to pass values on the stack now and the C++ side will just copy the string immediately.